base_layer/wallet/src/ffi.rs

```diff
@@ -120,284 +435,574 @@ pub unsafe extern "C" fn create_pending_transaction_outputs(
+    if wallet.is_null() {
+        return false;
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn generate_identity(wallet: *mut WalletFfi) -> *mut IdentityFfi {}
-// TODO C Destructuring methods for the IdentityFfi struct
+    if peer.is_null() {
+        return false;
+    }
 
-#[no_mangle]
-pub unsafe extern "C" fn add_base_node_peer(
-    wallet: *mut WalletFfi,
-    public_key: *const c_char,  // Byte[32] - PublicKey
-    secret_key: *const c_char,  // Byte[32] - SecretKey
-    net_address: *const c_char, // NetAddress
-) -> bool
-{
+    // (*wallet).addPeer((*peer));
+    return true;
 }
 
-#[no_mangle]
-pub unsafe extern "C" fn get_network_status(wallet: *mut WalletFfi) -> *mut NetworkStatusFfi {}
+//#[no_mangle]
+// pub unsafe extern "C" fn get_network_status(wallet: *mut Wallet) -> *mut NetworkStatusFfi {}
 // TODO C Destructuring methods for the NetworkStatusFfi struct
 
 #[no_mangle]
-pub unsafe extern "C" fn get_balance(wallet: *mut WalletFfi) -> c_ulonglong {}
+pub unsafe extern "C" fn wallet_get_balance(wallet: *mut Wallet) -> c_ulonglong {
+    //(*wallet).getBalance();
+    return 0;
+}
 
 // Create and send the first stage of a transaction to the specified wallet for the specified amount and with the
 // specified fee.
 #[no_mangle]
-pub unsafe extern "C" fn send_transaction(
-    wallet: *mut WalletFfi,
-    amount: c_ulonglong,       // MicroTari
-    fee_per_gram: c_ulonglong, // MicroTari
-    lock_height: c_ulonglong,  // u64
-    // Destination Node Peer
-    public_key: *const c_char,  // Byte[32] - PublicKey
-    net_address: *const c_char, // NetAddress
+pub unsafe extern "C" fn wallet_send_transaction(
+    wallet: *mut Wallet,
+    peer: *mut Peer,
+    transaction: *mut Transaction,
 ) -> bool
 {
-    // This function will need to check if the peer already exists and if not create it before sending
-    // A callback will be used by LibWallet to send the resultant data back to the client for storage.
+    if wallet.is_null() {
+        return false;
+    }
+
+    if transaction.is_null() {
+        return false;
+    }
+
+    if peer.is_null() {
+        return false;
+    }
+    //(*wallet).sendTransaction((*peer),(*transaction)) ? true : false
+    return true;
 }
 
 /// Cancel a pending outbound transaction so that the wallet will not complete and broadcast it if a reply is received
 #[no_mangle]
-pub unsafe extern "C" fn cancel_transaction(wallet: *mut WalletFfi, tx_id: c_ulonglong) -> bool {}
+pub unsafe extern "C" fn wallet_cancel_transaction(wallet: *mut Wallet, tr: *mut Transaction) -> bool {
+    if wallet.is_null() {
+        return false;
+    }
+
+    if tr.is_null() {
+        return false;
+    }
+
+    //(*wallet).cancelTransaction ((*tr)) ? true : false
+    return true;
+}
+
+// Callback Definition - Example
+
+// Will probably have to implement as a struct of callbacks in wallet, with wallet only calling the
```

Comment:
We will have a struct with all the calls back that monitors Channels for instructions from the wallet to call the relevant callback.